src/ziggurat/init.clj

```diff
@@ -143,12 +143,18 @@
                            (shutdown-agents))
             "Shutdown-handler")))
 
+(declare StreamRoute)
+
 (s/defschema StreamRoute
   (s/conditional
    #(and (seq %)
          (map? %))
-   {s/Keyword {:handler-fn (s/pred #(fn? %))
-               s/Keyword   (s/pred #(fn? %))}}))
+   {s/Keyword {:handler-fn                     (s/pred #(fn? %))
+               (s/optional-key :consumer-type) (s/enum :default :stream-joins)
+               (s/optional-key :input-topics)  (s/pred vector?)
+               (s/optional-key :output-topic)  (s/pred string?)
```

Comment:
I think we discussed this, we cannot have output-topics, as stream-routes will not be able to write to output topics once the message goes into the retry flow

Reply:
oh ok, I will remove it